os/fs/smartfs/smartfs_smart.c

```diff
@@ -298,8 +298,7 @@ static int smartfs_open(FAR struct file *filep, const char *relpath, int oflags,
 
 			/* At this point, either an available entry was found or a new one has been created */
 			newentry.firstsector = data_sector;
-
-			ret = smartfs_writeentry(fs, newentry, filename, SMARTFS_DIRENT_TYPE_FILE, mode, &sf->entry, parentdirsector);
+			ret = smartfs_writeentry(fs, newentry, filename, SMARTFS_DIRENT_TYPE_FILE, mode, &sf->entry, parentdirsector, sf);
```

Comment:
We'll write entry data in parent sector after first sector written.
So here we don't have to call writeentry.
We can call writeentry after sync_internal.
Please check below.
```
ret = smartfs_find_availableentry(fs, &parentdirsector, &sf->entry);
...
smartfs_wrle32(sf->entry->utc, fs->entry_seq++);
/* add below if we need it */
memset(direntry->name, 0, fs->fs_llformat.namesize + 1); 
strncpy(direntry->name, filename, fs->fs_llformat.namesize);
```

Reply:
Modified